MVVM/ViewModel/ZedMonitor.cs

```diff
@@ -0,0 +1,117 @@
+using Godot;
+using MQTTnet;
+using MQTTnet.Internal;
+using OpenCvSharp;
+using RoverControlApp.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text.Json;
+using RoverControlApp.MVVM.ViewModel;
+
+namespace RoverControlApp.MVVM.ViewModel;
+
+public partial class ZedMonitor : Panel
+{
+	//[Export]
+	double pitchDeg = 0;
+	//[Export]
+	double rollDeg = 0;
+
+    //Sprites for gyro visualisation
+    [Export]
+	Sprite2D pitchVisualisation;
+    [Export]
+    Sprite2D rollVisualisation;
+
+	//Labels for gyro display
+	[Export]
+	Label pitchDisplay;
+    [Export]
+	Label rollDisplay;
+
+	double QuatX, QuatY, QuatZ, QuatW;

```

Comment:
Is this really necessary here? Imo it should be passed as function argument below. Tip: U can use built-in Godot Quaterion.

Reply:
Another option: `Roll()` and `Pitch()` could be merged into `PullGyroscope()`, since they are both short and called right after.